snaps/clean.go

```diff
@@ -251,7 +251,7 @@ func examineSnaps(
 			return nil, isDirty, err
 		}
 
-		var hasDiffs bool
+		var needsUpdating bool
```

Comment:
this name still isn't really ideal, since arguably "needs sorting" is a subset of "needs updating" but I don't think that's too big of a problem - alternatively, it could be called something like "haveSnapshotsChanged"